From d722c6b6843a68aaaaa1984e821e8db003df6d5c Mon Sep 17 00:00:00 2001
From: Yuan Can <yuancan@huawei.com>
Date: Tue, 19 Jul 2022 11:36:08 +0800
Subject: [PATCH] mm/sharepool: Avoid NULL pointer dereference in
 mg_sp_group_add_task
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: d722c6b6843a68aaaaa1984e821e8db003df6d5c
Modified-by-SEL: No


ascend inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I5DS9S
CVE: NA

------------------------------------------------------

create_spg_node() may fail with NULL pointer returened, and in
the out_drop_spg_node path, the NULL pointer will be dereferenced
in free_spg_node().

Signed-off-by: Yuan Can <yuancan@huawei.com>
Reviewed-by: Weilong Chen <chenweilong@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 mm/share_pool.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/share_pool.c b/mm/share_pool.c
index edba445d4bbf..caf3e89b41c4 100644
--- a/mm/share_pool.c
+++ b/mm/share_pool.c
@@ -1283,7 +1283,7 @@ int mg_sp_group_add_task(int pid, unsigned long prot, int spg_id)
 	node = create_spg_node(mm, prot, spg);
 	if (unlikely(IS_ERR(node))) {
 		ret = PTR_ERR(node);
-		goto out_drop_spg_node;
+		goto out_drop_group;
 	}
 
 	/* per process statistics initialization */
-- 
2.33.0

